### ISSUE TITLE
`AllowUnsafeBlocks=true` makes all warning to error from v1.33.0

### ISSUE THREAD

**Issue by fatfatson** [OmniSharp/omnisharp-roslyn#1565]
too many  fake errors make real error hardly found -_-!
the earlier version doesn't have the problem.

**Comment by filipw** [comment]
can you please provide a repro example of what is not working? 
Since 1.32.20 OmniSharp correctly respects [TreatWarningsAsErrors](https://github.com/OmniSharp/omnisharp-roslyn/blob/master/CHANGELOG.md#13220---2019-06-03) so perhaps you are seeing that?

**Comment by fatfatson** [comment]
the example repo:  https://github.com/fatfatson/test-omnisharp

with the code:
```
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
            return;
            Console.WriteLine("Hello Again!");
        }
```

a warning is reported as:

![image](https://user-images.githubusercontent.com/5842874/61601446-f09de780-ac67-11e9-8238-b53260fbbfc0.png)


that's normal!

but if the `AllowUnsafeBlocks` build option is setting to `true` (which is neccesary for my project)
![image](https://user-images.githubusercontent.com/5842874/61601389-b3d1f080-ac67-11e9-9835-00a88db92e60.png)

the above warning will reported as error:
![image](https://user-images.githubusercontent.com/5842874/61601434-da902700-ac67-11e9-9c3e-0073f29f55c4.png)

I have found that from v1.33, all warnings are reported as error if `AllowUnsafeBlocks=true`,  no matter what `TreatWarningsAsErrors` is set.

**Comment by filipw** [comment]
thanks a lot for the repro. I see the problem now - this is indeed a regression. Will fix this.
